orchestra/cmds/components.py

```diff
@@ -70,39 +70,47 @@ def handle_components(args):
             if not fnmatch(branch, args.branch):
                 continue
 
-        installed_build = get_installed_build(component_name, config)
-        if args.installed and installed_build \
-                or args.not_installed and installed_build is None \
-                or not args.installed and not args.not_installed:
-
-            component_infos = []
-            if args.hashes:
-                component_infos.append(f"hash: {component.self_hash}")
-                component_infos.append(f"recursive hash: {component.recursive_hash}")
-            component_infos_s = stringify_infos(component_infos)
-
-            print(f"Component {component_name} {component_infos_s}")
-            for build_name, build in component.builds.items():
-                build_infos = []
-                if installed_build == build_name:
-                    build_infos.append("installed")
-                if build is component.default_build:
-                    build_infos.append("default")
-
-                if build.configure and args.deps:
-                    dependencies = [dep for dep in build.configure.dependencies]
-                    if dependencies:
-                        build_infos.append(f"config deps: {' '.join(d.name_for_components for d in dependencies)}")
-
-                if build.install and args.deps:
-                    dependencies = [dep for dep in build.install.dependencies]
-                    if dependencies:
-                        build_infos.append(f"install deps: {' '.join(d.name_for_components for d in dependencies)}")
-
-                build_infos_s = stringify_infos(build_infos)
-                print(f"  Build {build_name} {build_infos_s}")
-
-            print()
+        metadata = get_installed_metadata(component_name, config)
+        is_installed = metadata is not None
+        manually_installed = is_installed and metadata.get("manually_installed", False)
+        installed_build = metadata and metadata.get("build_name")
+        if args.installed and not is_installed:
+            continue
+        if args.not_installed and is_installed:
+            continue
+
+        component_infos = []
+        if args.hashes:
+            component_infos.append(f"hash: {component.self_hash}")
+            component_infos.append(f"recursive hash: {component.recursive_hash}")
+        component_infos_s = stringify_infos(component_infos)
+
+        print(f"Component {component_name} {component_infos_s}")
```

Comment:
Why `print` here? Shouldn't we use regular logging?
Can you do a project-wide check about `print` usage?

Reply:
The logging functions have decorations, colors, etc, commands which print informations directly (components, graph, dumpconfig, environment, ls) don't use them for this reason.
There was a usage in the clone command which I replaced with a call to the logger.

Reply:
OK, so this is "actual program results" let's say. So it makes sense to spit it on `stdout` instead of having it on `stderr` with frizzi e lazzi.